lib/rust/parser/src/syntax/token.rs

```diff
@@ -0,0 +1,316 @@
+//! Token implementation. Tokens are output from the lexer. Read the docs of the main module to
+//! learn more about the parsing process steps.
+
+use crate::prelude::*;
+
+use crate::source::span::Span;
+use crate::source::span::SpanRef;
+use crate::source::Code;
+use crate::source::Offset;
+
+use enso_shapely_macros::tagged_enum;
+
+
+
+// =============
+// === Token ===
+// =============
+
+/// A lexical token is a string with an assigned and thus identified meaning. Token remembers its
```

Comment:
This documentation belongs more to the module-level docstring.

Reply:
Moved and improved.